src/Microsoft.Management.Deployment/PackageManager.cpp

```diff
@@ -302,14 +308,77 @@ namespace winrt::Microsoft::Management::Deployment::implementation
                 context->Args.AddArg(Execution::Args::Type::Override, ::AppInstaller::Utility::ConvertToUTF8(options.ReplacementInstallerArguments()));
             }
         }
-        
-        AddPackageManifestToContext(packageVersionInfo, context);
 
-        // TODO: AdditionalPackageCatalogArguments is not currently supported by the underlying implementation.
+        // If the version of the package is specified use that, otherwise use the default.
+        Microsoft::Management::Deployment::PackageVersionInfo packageVersionInfo = GetPackageVersionInfo(package, options);
+        AddPackageManifestToContext(packageVersionInfo, context.get());
+
+        // Note: AdditionalPackageCatalogArguments is not needed during install since the manifest is already known so no additional calls to the source are needed. The property is deprecated.
         return context;
     }
 
-    winrt::Windows::Foundation::IAsyncOperationWithProgress<winrt::Microsoft::Management::Deployment::InstallResult, winrt::Microsoft::Management::Deployment::InstallProgress> GetInstallOperation(bool addToQueue, winrt::Microsoft::Management::Deployment::CatalogPackage package, winrt::Microsoft::Management::Deployment::InstallOptions options)
+    std::wstring GetQueueIdForInstall(winrt::Microsoft::Management::Deployment::CatalogPackage package, winrt::Microsoft::Management::Deployment::PackageCatalogInfo catalogInfo)
+    {
+        // Package Id + Catalog Id forms a unique identifier of a package for the purposes of the queue.
+        // Packages with the same id but from different catalogs can be installed at the same time.
+        std::wstring queueId;
+        queueId += package.Id();
+        queueId += std::wstring(L".");
+        queueId += catalogInfo.Id();
+        return std::move(queueId);
+    }
+
+    std::shared_ptr<Execution::OrchestratorQueueItem> GetQueueItemForPackage(winrt::Microsoft::Management::Deployment::CatalogPackage package, winrt::Microsoft::Management::Deployment::PackageCatalogInfo catalogInfo)
+    {
+        std::shared_ptr<Execution::OrchestratorQueueItem> queueItem = nullptr;
+        if (catalogInfo)
+        {
+            // If the caller has passed in the catalog they expect the package to have come from, then only look for an install from that catalog.
+            // Fail if they've used a catalog that doesn't have an Id. This can currently happen for Info objects that come from PackageCatalogReference objects for REST catalogs.
+            THROW_HR_IF(APPINSTALLER_CLI_ERROR_INVALID_CL_ARGUMENTS, catalogInfo.Id().size() == 0);
+            queueItem = Execution::ContextOrchestrator::Instance().GetQueueItem(GetQueueIdForInstall(package, catalogInfo));
+            return queueItem;
+        }
+
+        // If the caller has not specified the catalog, then check InstalledVersion,

```

Comment:
Can this comment be clarified to include the explicit reason that we need to check the InstalledVersion (as I had to re-run our conversation to remember why myself).  Basically, when it comes from the Installing catalog, this allows the active operation to be found.